### PR TITLE
Add ability to define custom environment variables

### LINKED ISSUE #76: Unable to set custom env or secrets
It's a common use case to be able to set custom env values for a deployment/stateful set but it doesn't seem like that is possible with this helm chart.

Would you accept a PR for changes to enable that? From both a list defined in the chart and from a secret. If you would, can you advise if there is a preferred format for that?

Use case for adding env to values.yaml config eg:

```yaml
deploy:
  container:
    envFromSecrets:
     - "mySecretNameContainingPostGresPassword"
    env:
      POSTGRES_URL: jdbc:postgresql://infinispan-postgres:5432/infinispan
      POSTGRES_USERNAME: infinispan

  # the infinispan config
  infinispan:
    cacheContainer:
      statistics: true
      caches:
        myTemplate:
          distributedCacheConfiguration:
            owners: ${env.STANDARD_OWNERS:2}
        myCacheName:
          distributedCache:
            configuration: myTemplate
      server:
        dataSources:
          - name: ds
            jndiName: 'jdbc/postgres'
            statistics: true
            connectionFactory:
              driver: org.postgresql.Driver
              url: '${env.POSTGRES_URL:jdbc:postgresql://infinispan-postgres:5432/infinispan}'
              username: '${env.POSTGRES_USERNAME:postgres}'
              password: '${env.POSTGRES_PASSWORD:mysecretpassword}'
```

We can then override the env for different environments with our own custom settings rather than having to update values throughout the charts.

### PR DESCRIPTION
**Pull request by gdsmith**
Closes #76

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -16,6 +16,21 @@ deploy:
...
+    #    valueFrom:
+    #      configMapKeyRef:
+    #        name: special-config
+    #        key: special.how
+    #  - name: SECRET_KEY
+    #    valueFrom:
+    #      secretKeyRef:
+    #        name: special-secret
+    #        key: special.how
+    extraEnv:
```
Comment (ryanemerson) [inline values.yaml:33]: There's an inconsistency here between the commented example and the default element, one is `env` the other is `extraEnv`. IMO `env` is better as it's more succinct and follows the k8s convention. From the user perspective, they are simply adding their env variables, the fact we add 3 of our own is irrelevant to them, so I don't think the added meaning of `extraEnv` is required.

**Review by ryanemerson — COMMENTED** [review]
Thanks! The PR works as expected, just the one comment.

File: values.yaml (hunk 1 above)
Comment (gdsmith) [reply values.yaml]: Yeah, I originally had it as `env` and switched it out. Switched it back now.

**Comment by ryanemerson** [comment]
Thanks @gdsmith

**Comment by gdsmith** [comment]
@ryanemerson thanks for merging, what is the release process?

**Comment by ryanemerson** [comment]
@gdsmith The PR has been merged, so 0.3.1 should be available now. Apologies for the delay.

**Pull request merged**